### ISSUE TITLE
-Wmissing-prototypes regression from llvm19

### ISSUE THREAD

**Issue by gbMattN** [llvm/llvm-project#133235]
With this reproducer

```
#include<cstdlib>

#pragma clang diagnostic error "-Wmissing-prototypes"

void operator delete( void* ptr, std::size_t size ) noexcept { /* Diagnostic expected */
    (void) ptr; /* avoid unused warning */
    (void) size;
}

int main(){

    int* p = new int;
    *p = 13;
    delete p;

    return 0;
}
```
In llvm18.1.8 you get the error
```
reproducer.cpp:6:6: error: no previous prototype for function 'operator delete' [-Werror,-Wmissing-prototypes]
    6 | void operator delete( void* ptr, std::size_t size ) noexcept { /* Diagnostic expected */
      |      ^
reproducer.cpp:6:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
    6 | void operator delete( void* ptr, std::size_t size ) noexcept { /* Diagnostic expected */
      | ^
      | static 
1 error generated.
```
However from llvm19.1.0, this is no longer caught.
This bug also happens if you define `void operator delete(void*)` without the sized version

**Comment by shafik** [comment]
CC @cor3ntin

**Comment by efriedma-quic** [comment]
I don't think this is a regression?  At least, it's only a regression in the sense that we turned on -fsized-deallocation by default.  Otherwise, the behavior is the same as it's always been: we don't warn on replaceable deallocation functions because they're implicitly declared by the compiler.  Which... at first glance, seems okay; not sure why we'd want to change the behavior here.
